hudi-common/src/main/java/org/apache/hudi/common/util/queue/DisruptorMessageQueue.java

```diff
@@ -60,6 +61,12 @@ public long size() {
 
   @Override
   public void insertRecord(I value) throws Exception {
+    if (!isStart) {
+      synchronized (this) {
+        wait();
+      }
+    }
```

Comment:
```suggestion
      synchronized (this) {
          If (!isStart) {
               wait();
        }
    }
```

We need to let synchronized cover isStart check in case that `isStart ` is changed after `if (!isStart)` but before `wait();`

But in this way, we will do synchronized check for each record ingestion right?

Reply:
Oops,forgot to add a double check here.

```java
if(!isStart) {
  synchronized(this) {
    if(!isStart) {
      wait();
    }
  }
}
```

Reply:
@boneanxs Disruptor is a *lock-free* message queue, let's avoid taking the locks in the hot-path to avoid defeating its purpose.

Instead we should do the following: 

1. We set the flag `started` in the `start` method
2. We assert in `insertRecord` that the queue is started
3. We fix the start-up sequence for Disruptor to make sure we start consumer before producer

Reply:
Sure, will change this.